code/ModelAdmin.php

```diff
@@ -238,6 +252,69 @@ public function getExportFields()
         return singleton($this->modelClass)->summaryFields();
     }
 
+    /**
+     * @deprecated 5.0
+     * @return \SilverStripe\ORM\Search\SearchContext
+     */
+    public function getSearchContext()
+    {
+        Deprecation::notice('5.0', 'Will be removed in favor of GridFieldFilterHeader in 5.0');
+
+        $gridField = $this->getEditForm()->fields()
+            ->fieldByName($this->sanitiseClassName($this->modelClass));
+
+        $filterHeader = $gridField->getConfig()
+            ->getComponentByType(GridFieldFilterHeader::class);
+
+        $context = $filterHeader
+            ->getSearchContext($gridField);
+
+        return $context;
+    }
+
+    /**
+     * Gets a list of fields that have been searched
+     *
+     * @deprecated 5.0
+     * @return ArrayList
+     */
+    public function SearchSummary()
+    {
+        Deprecation::notice('5.0', 'Will be removed in favor of GridFieldFilterHeader in 5.0');
+
+        $context = $this->getSearchContext();
+
+        return $context->getSummary();
+    }
+
+    /**
+     * Returns the search form
+     *
+     * @deprecated 5.0
+     * @return Form|bool
+     */
+    public function SearchForm()
```

Comment:
Following our discussion this morning ... there's no point whitelisting this in this context because the UI that would be using this has been removed.